pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotSegmentRestletResource.java

```diff
@@ -19,255 +19,409 @@
-  // TODO: 1. stateStr is not used
-  // TODO: 2. this method returns the map from server to segments instead of segment metadata
-  public String listMetadataForAllSegments(
+  @ApiOperation(value = "Get a map from server to segments hosted by the server (deprecated, use 'GET /segments/{tableName}/servers' instead)", notes = "Get a map from server to segments hosted by the server (deprecated, use 'GET /segments/{tableName}/servers' instead)")
+  public List<Map<String, String>> getServerToSegmentsMapDeprecated2(
       @ApiParam(value = "Name of the table", required = true) @PathParam("tableName") String tableName,
-      @ApiParam(value = "enable|disable|drop", required = false) @QueryParam("state") String stateStr,
-      @ApiParam(value = "realtime|offline", required = false) @QueryParam("type") String tableTypeStr) {
-    CommonConstants.Helix.TableType tableType = Constants.validateTableType(tableTypeStr);
-    return getInstanceToSegmentsMap(tableName, tableType);
+      @ApiParam(value = "MUST be null") @QueryParam("state") String stateStr,
+      @ApiParam(value = "OFFLINE|REALTIME") @QueryParam("type") String tableTypeStr)
+      throws JsonProcessingException {
+    return getServerToSegmentsMapDeprecated1(tableName, stateStr, tableTypeStr);
   }
 
   @GET
-  @Path("tables/{tableName}/segments/{segmentName}/metadata")
+  @Path("segments/{tableName}/crc")
+  @Produces(MediaType.APPLICATION_JSON)
+  @ApiOperation(value = "Get a map from segment to CRC of the segment (only apply to OFFLINE table)", notes = "Get a map from segment to CRC of the segment (only apply to OFFLINE table)")
+  public Map<String, String> getSegmentToCrcMap(
+      @ApiParam(value = "Name of the table", required = true) @PathParam("tableName") String tableName) {
+    String offlineTableName = getExistingTableNamesWithType(tableName, TableType.OFFLINE).get(0);
+    return _pinotHelixResourceManager.getSegmentsCrcForTable(offlineTableName);
+  }
+
+  @Deprecated
+  @GET
+  @Path("tables/{tableName}/segments/crc")
+  @Produces(MediaType.APPLICATION_JSON)
+  @ApiOperation(value = "Get a map from segment to CRC of the segment (deprecated, use 'GET /segments/{tableName}/crc' instead)", notes = "Get a map from segment to CRC of the segment (deprecated, use 'GET /segments/{tableName}/crc' instead)")
+  public Map<String, String> getSegmentToCrcMapDeprecated(
+      @ApiParam(value = "Name of the table", required = true) @PathParam("tableName") String tableName) {
+    return getSegmentToCrcMap(tableName);
+  }
+
+  @GET
+  @Path("segments/{tableName}/{segmentName}/metadata")
   @Produces(MediaType.APPLICATION_JSON)
-  @ApiOperation(value = "Lists one segment's metadata", notes = "Lists one segment's metadata")
-  public String listMetadataForOneSegment(
+  @ApiOperation(value = "Get the metadata for a segment", notes = "Get the metadata for a segment")
+  public List<Map<String, Object>> getSegmentMetadata(
```

Comment:
Not sure why we are returning a list of maps. A segment should only belong to one type of table. If the user specifies the table type, we can use that. Alternatively, we can derive the table type from the segment name and remove the table type argument altogether.

Reply:
Make sense, will remove the type parameter for APIs with segment name included.